platform/api/azure/instance.go

```diff
@@ -70,13 +71,19 @@ func (a *API) getVMParameters(name, userdata, sshkey, storageAccountURI string,
 			Sku:       &a.opts.Sku,
 			Version:   &a.opts.Version,
 		}
+		plan = &compute.Plan{
+			Publisher: imgRef.Publisher,
```

Comment:
Is this always valid given any disk URI or should it only be set when the else case above is hit?

Reply:
It's definitely valid here, but would also be valid if passed an uri to a gallery image disk. For images/VHD uris there is no attached plan information.

I don't know if there is a way to lookup the plan from an arbitrary disk uri.

Reply:
Above imfRef is set by a.opts.DiskURI and the plan values are empty then? Maybe it makes sense to have plan be nil in that case. If it's not a problem for the API, we can leave it as is.

Reply:
Ok, was just confused by the alignment here in GitHub^^